std/json.d

```diff
@@ -488,17 +522,24 @@ struct JSONValue
      * Hash syntax for json objects.
      * Throws: $(D JSONException) if $(D type) is not $(D JSON_TYPE.OBJECT).
      */
-    ref inout(JSONValue) opIndex(string k) inout pure @safe
+    ref inout(Value) opIndex(string k) inout pure @safe
     {
         auto o = this.objectNoRef;
-        return *enforce!JSONException(k in o,
-                                        "Key not found: " ~ k);
+        static if (hasMember!(Policy, "ObjectType"))
```

Comment:
DRY:
static if (hasMember!(Policy, "ObjectType"))
  auto ok=o.opIn(k);
else
  auto ok=k in o;
return *enforce!JSONException(ok, "Key not found: " ~ k);

Reply:
Just discovered that `opIn_r` was a thing...now that I know about it, this code only has the one case now :)